File: Factory/CPP/Botellas.cpp
#include <bits/stdc++.h>

using namespace std;

//Interfaz inicial para el resto de clases-----------------------------------------
class Botellas{
    private:
    
    public:
        //virtual define la clase abstracta obligatoria
        virtual void MiBotella()=0;
};



//primera clase heredada de la interfaz--------------------------------------------
class BotellasPet : public Botellas{
    private:

    public:
        virtual void MiBotella();
};

void BotellasPet::MiBotella(){
    cout<<"Pet"<<endl;
}



//Segunta clase heredada de la interfaz--------------------------------------------
class BotellasVidrio : public Botellas{
    private:

    public:
        virtual void MiBotella();
};

void BotellasVidrio::MiBotella(){
    cout<<"Vidrio"<<endl;
}

File: Factory/Python/Botellas.py
from abc import abstractclassmethod

class Botellas:
    @abstractclassmethod
    def MiBotella():
        pass

class BotellaVidrio(Botellas):
    @staticmethod
    def MiBotella():
        print("Vidrio")

class BotellaPet(Botellas):
    @staticmethod
    def MiBotella():
        print("Pet")